File: vibe-lang/benchmark/output/python-pygame/gemini/platformer_level.py
import pygame
import sys

pygame.init()
screen = pygame.display.set_mode((800, 600))
pygame.display.set_caption("Platformer")
clock = pygame.time.Clock()

# Player state
px, py = 100.0, 400.0
vx, vy = 0.0, 0.0
speed = 250.0
gravity = 600.0
jump_vel = -350.0
on_ground = False

# Platforms
plat_xs = [50.0, 300.0, 500.0, 200.0, 450.0]
plat_ys = [450.0, 380.0, 300.0, 220.0, 150.0]
plat_ws = [200.0, 150.0, 180.0, 120.0, 200.0]

font = pygame.font.Font(None, 36)

running = True
while running:
    dt = clock.tick(60) / 1000.0
    prev_y = py

    for event in pygame.event.get():
        if event.type == pygame.QUIT:
            running = False

    keys = pygame.key.get_pressed()
    
    # Horizontal movement
    vx = 0
    if keys[pygame.K_LEFT]: vx = -speed
    if keys[pygame.K_RIGHT]: vx = speed
    px += vx * dt

    # Vertical movement
    vy += gravity * dt
    if keys[pygame.K_UP] and on_ground:
        vy = jump_vel
        on_ground = False
    py += vy * dt

    # Collision detection
    on_ground = False
    
    # Platform collisions
    for i in range(len(plat_xs)):
        if vy >= 0 and prev_y + 20 <= plat_ys[i]:
            if py + 20 >= plat_ys[i] and px + 20 > plat_xs[i] and px < plat_xs[i] + plat_ws[i]:
                py = plat_ys[i] - 20.0
                vy = 0
                on_ground = True
    
    # Ground collision
    if py + 20 >= 580:
        py = 580 - 20.0
        vy = 0
        on_ground = True

    # Reset if fallen
    if py > 620:
        px, py = 100.0, 400.0
        vy = 0

    # Draw
    screen.fill((0, 0, 0))
    
    # Draw platforms
    for i in range(len(plat_xs)):
        pygame.draw.rect(screen, (255, 255, 255), (plat_xs[i], plat_ys[i], plat_ws[i], 12))
    
    # Draw player
    pygame.draw.rect(screen, (255, 255, 255), (int(px), int(py), 20, 20))
    
    # Draw text
    text = font.render("Use arrows + up to jump", True, (255, 255, 255))
    screen.blit(text, (10, 10))
    
    pygame.display.flip()

pygame.quit()
sys.exit()

File: benchmark/output/lua-love/gemini/platformer_level.lua
local px, py = 100, 400
local vx, vy = 0, 0
local prev_y = 400
local on_ground = false

local gravity = 600.0
local jump_vel = -350.0
local speed = 250.0

local plat_xs = {50, 300, 500, 200, 450}
local plat_ys = {450, 380, 300, 220, 150}
local plat_ws = {200, 150, 180, 120, 200}

function love.load()
    love.window.setMode(800, 600)
    love.window.setTitle("Platformer")
end

function love.update(dt)
    prev_y = py
    
    -- Horizontal movement
    vx = 0
    if love.keyboard.isDown("left") then vx = -speed end
    if love.keyboard.isDown("right") then vx = speed end
    px = px + vx * dt
    
    -- Vertical movement
    vy = vy + gravity * dt
    py = py + vy * dt
    
    -- Jump
    if on_ground and love.keyboard.isDown("up") then
        vy = jump_vel
        on_ground = false
    end
    
    -- Collision with platforms
    on_ground = false
    if vy >= 0 then
        for i = 1, #plat_xs do
            if prev_y + 20 <= plat_ys[i] and py + 20 >= plat_ys[i] and 
               px + 20 > plat_xs[i] and px < plat_xs[i] + plat_ws[i] then
                py = plat_ys[i] - 20.0
                vy = 0
                on_ground = true
            end
        end
        -- Ground collision
        if py + 20 >= 580 then
            py = 580 - 20
            vy = 0
            on_ground = true
        end
    end
    
    -- Reset if fallen
    if py > 620 then
        px, py = 100, 400
        vy = 0
    end
end

function love.draw()
    love.graphics.print("Use arrows + up to jump", 10, 10)
    
    -- Draw player
    love.graphics.rectangle("fill", px, py, 20, 20)
    
    -- Draw platforms
    for i = 1, #plat_xs do
        love.graphics.rectangle("fill", plat_xs[i], plat_ys[i], plat_ws[i], 12)
    end
    
    -- Draw ground
    love.graphics.rectangle("fill", 0, 580, 800, 20)
end